### PR TITLE
fix: List widget issues in Auto Layout

### LINKED ISSUE #22833: [Bug][List] when we resize the list, the image widget disappears for a while second -item container onwards
https://www.loom.com/share/e5dbe660715144348439d237e39ca148

### LINKED ISSUE #23178: [Bug][List] With scroll contents ON in the list item container, the last widget crops
Image

### LINKED ISSUE #23181: [Bug][List] Text widget within List is having overflow property available

### PR DESCRIPTION
**Pull request by aswathkk**
## Description
- Removes overflow property from Text widgets within List
- Fix issue where the image widget disappears from List widget when resizing
- Fix issue where last widget crops when scrolling enabled in List widget container

#### PR fixes following issue(s)
Fixes #23181
Fixes #22833
Fixes #23178

#### Type of change
- Bug fix (non-breaking change which fixes an issue)

- [ ] Manual
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Test-plan-implementation#speedbreaker-features-to-consider-for-every-change) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans/_edit#areas-of-interest)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

File: app/client/src/widgets/CheckboxGroupWidget/component/index.tsx [hunk 1]
```diff
@@ -152,14 +149,13 @@ export interface CheckboxGroupComponentProps extends ComponentProps {
   labelTooltip?: string;
   accentColor: string;
   borderRadius: string;
-  isAutoLayout: boolean;
+  widgetId: string;
```
Comment (aswathkk) [inline app/client/src/widgets/CheckboxGroupWidget/component/index.tsx:152]: widgetId is available in ComponenetProps. Check if we need to remove this or not

**Review by marks0351 — COMMENTED** [review]
Changes look good to me. Pls get this verified by someone on the app viewers team, just to be sure.

**Review by marks0351 — DISMISSED** [review]

**Review by marks0351 — APPROVED** [review]

**Comment by kamakshibhat-appsmith** [comment]
Unable to place widgets and see drop targets when widget is dragges to a blank area (increase the height of list item and try dragging and releasing widget in the blank space )in List item container 

www.loom.com/share/0c41037da9ab4a1885bae9f3bb8adefd
Let me know if you want to fix it in this PR only or take it up separately
@aswathkk

**Comment by aswathkk** [comment]
@kamakshibhat-appsmith I will take that up separately.

**Pull request merged**